lib/tasks/data/remove_transient_data.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class RemoveTransientData
+  # This model lets us operate on the sessions DB table using ActiveRecord's
+  # methods within the scope of this service. This relies on the AR's
+  # convention where a Session model maps to a sessions table.
+  class Session < ActiveRecord::Base
+  end
+
+  def call
+    Rails.logger.info("RemoveTransientData: processing")
+
+    Spree::StateChange.delete_all("created_at < '#{retention_period}'")
```

Comment:
```suggestion
    Spree::StateChange.where("created_at < ?", retention_period).delete_all
```

Reply:
Given `Spree::StateChange` has no callbacks, I would prefer getting the performance boost of `delete_all`. Instantiating a potentially large list of models will have its cost.

Reply:
I'm still suggesting to use `delete_all`. The `where` clause is just a bit neater.

Reply:
right, I got confused with the `where` and thought it would instantiate the records. Yes, it won't.